src/ethereum_spec_tools/evm_tools/t8n/__init__.py

```diff
@@ -228,12 +228,25 @@ def _run_blockchain_test(self, block_env: Any, block_output: Any) -> None:
                 self.logger.warning(f"Transaction {i} failed: {e!r}")
 
         if not self.fork.is_after_fork("ethereum.paris"):
-            self.fork.pay_rewards(
-                block_env.state,
-                block_env.number,
-                block_env.coinbase,
-                self.env.ommers,
-            )
+            if self.options.state_reward is not None:
```

Comment:
Another way to do this might be to also expose the forks `BLOCK_REWARD` constant and define a method in the `T8N` class with this whole logic. Say `def pay_block_rewards(...)` and then do

```
if not self.fork.is_after_fork("ethereum.paris"):
    if self.options.state_reward is None:
        pay_block_rewards(self.fork.BLOCK_REWARD, ...)
    elif self.options.state_reward != -1
        pay_block_rewards(self.options.state_reward, ...)
```

Or something along these lines